src/mlstacks/utils/terraform_utils.py

```diff
@@ -378,13 +378,17 @@ def tf_client_apply(
             skip_plan=not debug,
         )
     except python_terraform.TerraformCommandError as e:
-        if "The specified location constraint is not valid" in e.out:
+        if e.out and "The specified location constraint is not valid" in e.out:
```

Comment:
I'm sorry don't get why are we adding the check for `e.out` if the following can already validate the check.
```
if "The specified location constraint is not valid" in e.out:
```

Reply:
The problem is that sometimes e.out is `None`, so if that is the case then trying to evaluate `if ... in e.out` will fail since `None` is not an iterable.